v2.0/sql-name-resolution.md

```diff
@@ -4,45 +4,260 @@ summary: Table and function names can exist in multiple places. Resolution decid
 toc: false
 ---
 
-A SQL client can have access to multiple databases side-by-side. The
-same table name (e.g., `orders`) can exist in multiple
-databases. When a query specifies a table name without a database
-name (e.g., `SELECT * FROM orders`), how does CockroachDB know
-which `orders` table is being considered?
+<span class="version-tag">Changed in v2.0:</span> A query can specify a table name without a database or schema name (e.g., `SELECT * FROM orders`). How does CockroachDB know which `orders` table is being considered and in which schema?
 
-This page details how CockroachDB performs **name resolution** to answer
-this question.
+This page details how CockroachDB performs **name resolution** to answer this question.
 
 <div id="toc"></div>
 
-## Overview
+## Logical Schemas And Namespaces
 
-The following **name resolution algorithm** is used both to determine
-table names in [table expressions](table-expressions.html) and
-function names in [value expressions](sql-expressions.html):
+<span class="version-tag">New in v2.0.</span>
 
-- If the name is *qualified* (i.e., the name already tells where to look), use this information.
-  For example, `SELECT * FROM db1.orders` will look up "`orders`" only in `db1`.
-- If the name is *unqualified*:
-  - Try to find the name in the "default database" as set by [`SET DATABASE`](set-vars.html).
-  - Try to find the name using the [search path](#search-path).
-  - If the name is not found, produce an error.
+A CockroachDB cluster can store multiple databases, and each database
+can store multiple tables/views/sequences. This **two-level structure
+for stored data** is commonly called the "logical schema" in
+relational database management systems.
 
-## Search Path
+Meanwhile, CockroachDB aims to provide compatibility with PostgreSQL
+client applications and thus supports PostgreSQL's semantics for SQL
+queries. To achieve this, CockroachDB supports a **three-level
+structure for names**. This is called the **naming hierarchy**.
```

Comment:
Since you've gone with quotes for "logical schema", I'll do the same here. The Docs team needs to work out more detailed style conventions for bold/italics/etc. But I generally lean away from italics; they're not very effective visually.

Reply:
Done.